src/modules/synchronizer/GitHubSynchronizer.js

```diff
@@ -0,0 +1,170 @@
+import User from 'server/models/User'
+import UserOrganizationRight from 'server/models/UserOrganizationRight'
+import UserRepositoryRight from 'server/models/UserRepositoryRight'
+
+const OWNER_ORGANIZATION = 'Organization'
+const OWNER_USER = 'User'
+
+class GitHubSynchronizer {
+  constructor(synchronization) {
+    this.synchronization = synchronization
+
+    this.github = new GitHubAPI({
+      debug: config.get('env') === 'development',
+    })
+    this.github.authenticate({
+      type: 'oauth',
+      token: synchronization.user.accessToken,
+    })
+  }
+
+  async synchronizeAuthUser() {
+    const githubUser = await this.github.users.get({})
+    await this.synchronization.user.$query().patch({
+      githubId: githubUser.id,
+      name: githubUser.name,
+      email: githubUser.email,
+    })
+  }
+
+  async synchronizeRepositories({ page = 1 } = {}) {
+    const githubRepositories = await this.github.repos.getAll({ page, per_page: 100 })
+
+    const [organizationIdByRepositoryId, userIdByRepositoryId] = await Promise.all([
+      this.synchronizeOwners(githubRepositories, OWNER_ORGANIZATION),
+      this.synchronizeOwners(githubRepositories, OWNER_USER),
+    ])
+
+    await Promise.all(githubRepositories.map(async (githubRepository) => {
+      const data = {
+        githubId: githubRepository.id,
+        name: githubRepository.name,
+        organizationId: organizationIdByRepositoryId[githubRepository.id],
+        userId: userIdByRepositoryId[githubRepository.id],
+      }
+
+      let [repository] = await Repository.query().where({ githubId: githubRepository.id })
+
+      if (repository) {
+        await repository.$query().patch(data)
+      } else {
+        repository = await Repository.query().insert({
+          ...data,
+          enabled: false,
+        })
+      }
+
+      const [userRepositoryRight] = await UserRepositoryRight.query().where({
+        userId: this.synchronization.user.id,
+        repositoryId: repository.id,
+      })
+
+      if (!userRepositoryRight) {
+        await UserRepositoryRight.query().insert({
+          userId: this.synchronization.user.id,
+          repositoryId: repository.id,
+        })
+      }
+    }))
+
+    if (this.github.hasNextPage(githubRepositories)) {
+      await this.synchronizeRepositories({ page: page + 1 })
+    }
+  }
+
+  async synchronizeOwners(githubRepositories, type) {
+    const githubOwners = githubRepositories
+      .reduce((githubOwners, githubRepository) => {
+        if (githubRepository.owner.type !== type) {
+          return githubOwners
+        }
+
+        let githubOwner = githubOwners.find(({ id }) => id === githubRepository.owner.id)
+
+        if (!githubOwner) {
+          githubOwner = githubRepository.owner
+          githubOwners.push(githubRepository.owner)
+        }
+
+        return githubOwners
+      }, [])
+
+    let owners
+
+    switch (type) { // eslint-disable-line default-case
```

Comment:
Why disabling `// eslint-disable-line default-case` here?
Option 1. We add a default case
Option 2. We disable that rule globally.